game.py

```diff
@@ -0,0 +1,126 @@
+from pieces import *
+from board import *
+from ascii import *
+from os import system, name
+import time
+
+#need to fix knight moves and make it so it doesn't change players if an invalid move for a piece is picked
+
+def clear():
+    # for windows
+    if name == 'nt':
+      system('cls')
+    # for mac and linux(here, os.name is 'posix')
+    else:
+      system('clear')
+
+def turn(board, last_move):
+  while True:
+    clear()
+    if last_move is not None: print(last_move)
+    print("\n\n")
+    board.update_moves()
+    if board.white_to_move == True:
+      print("White's turn to move.\n\n")
+      current_player = "White"
+      opposing_player = "Black"
+    else:
+      print("Black's turn to move.\n\n")
+      current_player = "Black"
+      opposing_player = "White"
+    board.ascii.draw_board(board)
+    print("Enter ? for a list of commands or enter your move (e.g. 'move e7 to e5' or 'move g8 to f6').")
+    user_input = input()
+    if user_input == "?":
+      print("List of commands\n\n'?' for a list of commands\n'q' quits to the menu\n'draw' to request a draw\n'resign' to resign\n'save' to save the current state of the board\n'undo' to request an undo of the last move\n'board' to view the board\n'move' (e.g. 'move e7 to e5')\n")
+    elif user_input == "q":
+      return "break"
+    elif user_input == "resign":
+      while True:
+        print("Are you sure you want to resign? (y/n)")
+        resignation = input()
+        if resignation == "y":
+          print(current_player + " has resigned. " + opposing_player + " wins!")
+          time.sleep(3)
+          return "break"
+        if resignation == "n":
+          pass
+    elif user_input == "draw":
+      pass
+    elif user_input == "save":
+      pass
+    elif user_input == "undo":
+      pass
+    elif user_input == "board":
+      board.ascii.draw_board(board)
+    elif user_input.startswith("move ") and len(user_input) == 13 and user_input[5:7] in board.algebraic_squares and user_input[11:13] in board.algebraic_squares:
```

Comment:
Since the set of squares is static and never changes, I'm not sure how useful it is to have pre-calculated `algebraic_squares` and to force this code here to think in terms of whether this input belongs in that set.

Instead, why don't we make a new method and have it be the board's job to answer the question "is this a valid square?". And in that method, we have lots of choices, like pulling apart the string to see if the first letter is a valid column and the second a valid row, or just passing this user input to the index method and seeing if it produces a valid index into our list of squares.

Reply:
This is a good idea. Thanks; will do.